Libraries/LogHelpers.js

```diff
@@ -38,20 +38,19 @@ function get_version_and_board(log) {
         if ("FV" in VER) {
             filter_version = VER.FV[0]
         }
-    }
-
-    if ('MSG' in log.messageTypes) {
-        const MSG = log.get("MSG")
 
-        if (build_names[build_type] && !fw_string.startsWith(build_names[build_type])) {
+        if ((build_names[build_type] != null) && !fw_string.startsWith(build_names[build_type])) {
             // If the log is from OEM-customized firmware with an
-            // AP_CUSTOM_FIRMWARE_STRING, append the base firmware info to match
-            // how it will appear in the MSGs. Otherwise, os_string and
-            // flight_controller will be returned undefined.
-            const { Maj = [0], Min = [0], Pat = [0] } = log.get("VER")
+            // AP_CUSTOM_FIRMWARE_STRING, append the base firmware info
+            // This also means it to matches how it will appear in the MSGs.
+            // So extraction of os_string and fight_controller will be returned undefined.
+            const { Maj = ['?'], Min = ['?'], Pat = ['?'] } = log.get("VER")
```

Comment:
This might just be me not understanding the comment, but I feel this would be slightly better phrasing:

```suggestion
            // AP_CUSTOM_FIRMWARE_STRING, append the base firmware info.
            // This also means it will match how it appears in the MSGs,
            // so os_string and fight_controller will not be returned undefined.
            const { Maj = ['?'], Min = ['?'], Pat = ['?'] } = log.get("VER")
```

Otherwise, if you don't think so, I'd say this is ready to go.